src/Headless/CMakeLists.txt

```diff
@@ -0,0 +1,71 @@
+cmake_minimum_required(VERSION 3.13)
+if(APPLE)
+    cmake_policy(SET CMP0042 NEW)
+endif(APPLE)
+
+set(ra_headless_target Headless)
+list(APPEND CMAKE_MESSAGE_INDENT "[${ra_headless_target}] ")
+project(${ra_headless_target} LANGUAGES CXX VERSION ${Radium_VERSION})
+# -----------------------------------------------------------------------------------
+# lib headless configuration
+# -----------------------------------------------------------------------------------
+# find required dependencies
+find_package(glfw3 3.2 QUIET)
+if(NOT glfw3_FOUND)
+    # do not configure Radium headless if dependency is not found
+    message(WARNING "glfw3 package not found. Headless component is disabled")
+else()
+    # configure the headless component for build and install as component of the Radium package
+    message(STATUS "Headless component is enabled")
+    # set source files
+    set(CLI11_headers
+        CLI/App.hpp
+        CLI/CLI.hpp
+        CLI/Config.hpp
+        CLI/ConfigFwd.hpp
+        CLI/Error.hpp
+        CLI/Formatter.hpp
+        CLI/FormatterFwd.hpp
+        CLI/Macros.hpp
+        CLI/Option.hpp
+        CLI/Split.hpp
+        CLI/StringTools.hpp
+        CLI/Timer.hpp
+        CLI/TypeTools.hpp
+        CLI/Validators.hpp
+        CLI/Version.hpp
+    )
+
+    set(${ra_headless_target}_sources CLIBaseApplication.cpp CLIViewer.cpp
```

Comment:
Shouldn't we use `cmake/filelistHeadless.cmake` instead ?

Reply:
I'm not a great fan of the pattern 'cmake/filelist.cmake' which I found harder to maintain than direct list of files and that prevent some IDE facilities (when adding a new class, the filelist must be updated manually). As this is not part of the [Coding conventions for Radium](https://storm-irit.github.io/Radium-Engine/release-candidate/develCodingConvention.html) I did this. But feel free to modify and commit/push on this branch.